README.md

```diff
@@ -269,6 +269,16 @@ A `JsonApiClient::ResultSet` object should be paginatable with both `kaminari` a
 Person.where(name: 'Jeff').all
 ```
 
+## Headers
+
+You can specify the headers to send with all requests within a block with `with_headers`.
```

Comment:
After some discussion with @justin-yesware offline, I'm inclined to merge this and change the implementation to use a thread local variable in a separate pull. The syntax here won't change, so the documentation will stay the same.